.github/workflows/deploy.yaml

```diff
@@ -49,4 +76,138 @@ jobs:
+    # so that we can authenticate with AWS
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Terraform
+        uses: ./.github/actions/setup-terraform
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
+          aws-account-id: ${{ secrets.AWS_ACCOUNT_ID }}
+          batch-container-image-name: ${{ needs.publish-docker-image.outputs.image-name }}
+          role-duration-seconds: 14400  # Worst-case time for a full model run
+
+      - name: Validate Terraform config
+        run: terraform validate
+        working-directory: terraform
+        shell: bash
+
+      - name: Apply Terraform changes
+        run: terraform apply -auto-approve
+        working-directory: terraform
+        shell: bash
+
+      - name: Submit new Batch job
+        id: submit-job
+        run: |
+          # Use terraform-bin to disable the wrapper script installed by
+          # the setup-terraform action, which adds extra context to
+          # `terraform output` calls
+          BATCH_JOB_NAME="$(terraform-bin output -raw batch_job_name)"
+          BATCH_JOB_DEFINITION="$(terraform-bin output -raw batch_job_definition_arn)"
+          BATCH_JOB_QUEUE="$(terraform-bin output -raw batch_job_queue_arn)"
+
+          BATCH_JOB_DETAILS=$(\
+            aws batch submit-job \
+              --job-name "$BATCH_JOB_NAME" \
+              --job-definition "$BATCH_JOB_DEFINITION" \
+              --job-queue "$BATCH_JOB_QUEUE" \
+          )
+          BATCH_JOB_ID=$(echo $BATCH_JOB_DETAILS | jq -r ".jobId")
+          echo "batch-job-id=$BATCH_JOB_ID" >> "$GITHUB_OUTPUT"
+        shell: bash
+        working-directory: terraform
+
+      - name: Wait for Batch job to complete
+        run: |
+          LOOP_COUNTER=0
+          JOB_HAS_STARTED=0
+          while true; do
+            echo "Getting status of job $BATCH_JOB_ID"
+            JOB_DESCRIPTIONS=$(aws batch describe-jobs --jobs "$BATCH_JOB_ID")
+
+            JOB_LIST=$(echo "$JOB_DESCRIPTIONS" | jq -r '.jobs')
+            if [[ "$JOB_LIST" == "[]" ]]; then
+              echo "Unexpected empty response from aws batch describe-jobs"
+              exit 1
+            fi
+
+            JOB_STATUS=$(echo "$JOB_DESCRIPTIONS" | jq -r '.jobs[0].status')
+            echo "Job status is $JOB_STATUS"
+
+            JOB_LOG_STREAM_NAME=$(\
+              echo "$JOB_DESCRIPTIONS" | \
+              jq -r '.jobs[0].container.logStreamName' \
+            )
+            # Any slashes in the log stream name need to be urlencoded
+            JOB_LOG_URL="${BATCH_JOB_LOG_URL_PREFIX}/${JOB_LOG_STREAM_NAME//\//%2F}"
+
+            case "$JOB_STATUS" in
+              "RUNNING")
+                if [[ "$JOB_HAS_STARTED" == "0" ]]; then
+                  # Only print the logs on the first loop, to keep output clean
+                  echo "See logs: $JOB_LOG_URL"
```

Comment:
See here for an example of how this looks when printed to the console: https://github.com/ccao-data/model-res-avm/actions/runs/6711601875/job/18239357137#step:7:112

Reply:
Both suggestions done in 2a85c146466071bfde2d319b2674c074ca2863ff. I set the new run interval to 5min, and split out the polling step into two steps: one that waits for job startup and prints a link to the logs once the job is running, and another that waits for job completion. Not exactly the same thing as a `link outputs` step, since the reader will have to scroll to the bottom of the step to get the link, but it should be easier than skimming through loop iterations to find the link and it makes the abstraction cleaner.